src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/WaitForIssueUpdateStep.java

```diff
@@ -0,0 +1,148 @@
+package org.thoughtslive.jenkins.plugins.jira.steps;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.jenkinsci.plugins.workflow.support.actions.PauseAction;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.thoughtslive.jenkins.plugins.jira.util.JiraStepDescriptorImpl;
+import org.thoughtslive.jenkins.plugins.jira.webhook.JiraIssueChangeListener;
+import org.thoughtslive.jenkins.plugins.jira.webhook.JiraWebHook;
+
+import com.google.common.collect.ImmutableSet;
+
+import hudson.Extension;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+/**
+ * Step to wait for an issue field to be updated.
+ *
+ * @author Ludovic Cintrat
+ */
+public class WaitForIssueUpdateStep extends Step implements Serializable {
+
+  private static final long serialVersionUID = -6061636623882913257L;
+
+  private final String idOrKey;
+
+  private final String field;
+
+  private final String[] fieldValues;
+
+  @DataBoundConstructor
+  public WaitForIssueUpdateStep(final String idOrKey, final String field, final String[] fieldValues) {
+
+    this.idOrKey = idOrKey;
+    this.field = field;
+    this.fieldValues = Arrays.copyOf(fieldValues, fieldValues.length);
+  }
+
+  @Override
+  public StepExecution start(StepContext context) throws Exception {
+    return new Execution(this, context);
+  }
+
+  static class Execution extends StepExecution implements JiraIssueChangeListener {

```

Comment:
Lets extend this from `JiraStepExecution` so that we can get the site name to check for.

Reply:
You have to check site name iff project keys are the same on different Jira instances, which is a bad practice. IMHO wait for a user need to implement this in another PR.

Reply:
I think probably we need to update the documentation to not to mention the site details as those are what we are coding for. 

But in general I would vote for extending it now and check I came across various implementation doing this at few clients.